backend/bottari/src/main/java/com/bottari/teambottari/service/TeamBottariService.java

```diff
@@ -94,12 +96,74 @@ public Long create(
         }
     }
 
+    @Transactional
+    public void exit(
+            final Long id,
+            final String ssaid
+    ) {
+        final TeamBottari teamBottari = teamBottariRepository.findById(id)
+                .orElseThrow(() -> new BusinessException(ErrorCode.TEAM_BOTTARI_NOT_FOUND));
+        final List<TeamMember> allTeamMembers = teamMemberRepository.findAllByTeamBottariId(teamBottari.getId());
+        final TeamMember exitTeamMember = findExitTeamMember(ssaid, allTeamMembers);
+        final List<TeamAssignedItem> teamAssignedItems = teamAssignedItemRepository.findAllByTeamMemberId(exitTeamMember.getId());
+        deleteItemsByTeamMember(exitTeamMember);
+        deleteOrphanAssignedItemInfos(teamAssignedItems);
+        final List<TeamMember> remainMembers = allTeamMembers.stream()
+                .filter(teamMember -> !teamMember.getMember().isSameBySsaid(ssaid))
+                .collect(Collectors.toList());
+        transferOwnershipIfNeeded(remainMembers, teamBottari, exitTeamMember);
+        teamMemberRepository.deleteById(exitTeamMember.getId());
+        deleteTeamBottariIfEmpty(remainMembers, teamBottari);
+    }
+
+    private void deleteItemsByTeamMember(final TeamMember exitTeamMember) {
+        teamPersonalItemRepository.deleteByTeamMemberId(exitTeamMember.getId());
+        teamSharedItemRepository.deleteByTeamMemberId(exitTeamMember.getId());
+        teamAssignedItemRepository.deleteByTeamMemberId(exitTeamMember.getId());
+    }
+
+    private void deleteTeamBottariIfEmpty(
+            final List<TeamMember> remainMembers,
+            final TeamBottari teamBottari
+    ) {
+        if (remainMembers.isEmpty()) {
+            teamSharedItemInfoRepository.deleteByTeamBottariId(teamBottari.getId());
+            teamBottariRepository.deleteById(teamBottari.getId());
+        }
+    }
+
+    private void transferOwnershipIfNeeded(
+            final List<TeamMember> remainMembers,
+            final TeamBottari teamBottari,
+            final TeamMember exitTeamMember
+    ) {
+        if (!remainMembers.isEmpty() && teamBottari.isOwner(exitTeamMember.getMember())) {
+            remainMembers.sort(Comparator.comparing(TeamMember::getCreatedAt));
+            final TeamMember newOwner = remainMembers.getFirst();
+            teamBottari.changeOwner(newOwner.getMember());
+        }
+    }
```

Comment:
```suggestion
    private void transferOwnershipIfNeeded(
            final List<TeamMember> remainMembers,
            final TeamBottari teamBottari,
            final TeamMember exitTeamMember
    ) {
        // 방장 변경이 필요 없는 경우, 메서드 초입에서 바로 리턴
        if (!teamBottari.isOwner(exitTeamMember.getMember()) || remainMembers.isEmpty()) {
            return;
        }
        // 방장 변경 로직 수행
        remainMembers.stream()
                .min(Comparator.comparing(TeamMember::getCreatedAt)) // min() 가장 오래된 유저 1개만
                .map(TeamMember::getMember) // 멤버로 꺼내서
                .ifPresent(teamBottari::changeOwner); // 방장 변경
    }
```

메서드명으로는 필요한 경우 방장 바꾸기,
로직으로는 방장이 나갈 경우에 가장 오래된 멤버에게 방장을 위임한다는 것을 잘 확인했어요 👍 

이 부분을 `얼리 리턴 + 스트림`을 활용하는건 어때요?

방장 위임이 필요 없는 예외 케이스를 메서드 맨 위에서 먼저 처리해주면, 
그 아래의 핵심 로직(새 방장을 찾는 부분)에 더 집중해서 코드를 읽을 수 있을 것 같아요.

해당 변경으로 깨진 테스트는 없습니다!  
<img width="214" height="24" alt="Image" src="https://github.com/user-attachments/assets/5f1e2da3-60d5-4ccc-a676-b8813b50ab5a" />

Reply:
아 좋네요! 훨씬 잘 읽히네요 수정해볼게요~